### ISSUE TITLE
[Bug]: Insertion after deletion of the data of same primary key may be invalid

### ISSUE THREAD

**Issue by xuefeimiaoao** [milvus-io/milvus#21446]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.2
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): milvus-sdk-java 2.2.1
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Firstly, i inserted three values into milvus:
`    val fields: util.List[InsertParam.Field] = new util.ArrayList[InsertParam.Field]`
`    val ids = util.Arrays.asList(100000L, 100001L, 100002L)`
`    val ages = util.Arrays.asList(99, 98, 97)`;

Then, i deleted those values:
`example.delete(collectionName, "id in [100000,100001,100002]")`;

Then, inserted three values:
`    val fields02: util.List[InsertParam.Field] = new util.ArrayList[InsertParam.Field]`
`    val ids02 = util.Arrays.asList(100000L, 200001L, 200002L)`
`    val ages02 = util.Arrays.asList(199, 198, 197)`;

Finally, when i query all data:
`example.query(
      collectionName,
      "age > 0",
      java.util.Arrays.asList("id", "age")
    )`;
The data `id in [100001,100002`] are deleted successfully.
However, the data of `id=100000,age=199` is missing, with the initial data `id=100000,age=99` left in milvus, which should have been updated.

Here is the query result finally:
![result](https://user-images.githubusercontent.com/76927591/209918207-2fd09289-a66a-4ba7-8913-5b0b30396874.png)

**Comment by xiaofan-luan** [comment]
/assign @yhmo 

could you help on investigating?

**Comment by yanliang567** [comment]
/assign @binbinlv 
sounds like a case we shall cover.

**Comment by xiaofan-luan** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Environment
> ```gfm
> - Milvus version: 2.0.2
> - Deployment mode(standalone or cluster): standalone
> - MQ type(rocksmq, pulsar or kafka):    
> - SDK version(e.g. pymilvus v2.0.0rc2): milvus-sdk-java 2.2.1
> - OS(Ubuntu or CentOS): 
> - CPU/Memory: 
> - GPU: 
> - Others:
> ```
> 
> ### Current Behavior
> Firstly, i inserted three values into milvus: ` val fields: util.List[InsertParam.Field] = new util.ArrayList[InsertParam.Field]` ` val ids = util.Arrays.asList(100000L, 100001L, 100002L)` ` val ages = util.Arrays.asList(99, 98, 97)`;
> 
> Then, i deleted those values: `example.delete(collectionName, "id in [100000,100001,100002]")`;
> 
> Then, inserted three values: ` val fields02: util.List[InsertParam.Field] = new util.ArrayList[InsertParam.Field]` ` val ids02 = util.Arrays.asList(100000L, 200001L, 200002L)` ` val ages02 = util.Arrays.asList(199, 198, 197)`;
> 
> Finally, when i query all data: `example.query( collectionName, "age > 0", java.util.Arrays.asList("id", "age") )`; The data `id in [100001,100002`] are deleted successfully. However, the data of `id=100000,age=199` is missing, with the initial data `id=100000,age=99` left in milvus, which should have been updated.
> 
> Here is the query result finally: ![result](https://user-images.githubusercontent.com/76927591/209918207-2fd09289-a66a-4ba7-8913-5b0b30396874.png)
> 
> ### Expected Behavior
> _No response_
> 
> ### Steps To Reproduce
> _No response_
> 
> ### Milvus Log
> _No response_
> 
> ### Anything else?
> _No response_

is this Milvus 2.0.2 or 2.2.2?

**Comment by xiaofan-luan** [comment]
We fixed many delete related issues in 2.1 and 2.2 so I highly recommend you to upgrade to 2.2.2

**Comment by xuefeimiaoao** [comment]
> 

It looks good when i upgrade milvus server to 2.2.2 and add one vector column to my table. @xiaofan-luan
